daffodil-lib/src/main/scala/org/apache/daffodil/xml/DaffodilXMLLoader.scala

```diff
@@ -237,12 +237,7 @@ class DFDLCatalogResolver private ()
     optURI match {
       case None => null
       case Some(uri) => {
-        val resourceAsStream =
-          try {
-            uri.toURL.openStream() // This will work.
-          } catch {
-            case _: java.io.IOException => Assert.invariantFailed("found resource but couldn't open")
-          }
+        val resourceAsStream = uri.toURL.openStream()
```

Comment:
You are in daffodil-lib code, which defines the ThrowsSDE stuff. You should be able to actually report an SDE from this if resourceAsStream comes back null, or catch the IOException, and nest whatever the I/O exception was as the SDE cause.  Perhaps that happens at a high level of catch structure?
The key is that this cause here gets captured.

Reply:
The DFDLCatalogResolver, where this code lives, doesn't mixin ThrowsSDE, so that would be a change to allow it to do that. In fact, this function actually gets called from deep within Xerces, and looking at the Xerces code, I think this function actually shouldn't ever throw an exception, including an SDE or IOException. I believe it should should return either an Input or null. And if null, Xerces then does some stuff to figure out why it failed and throw an actual exception, which Daffodil already handles and converts to an SDE appropriately.

So I think we should actually just catch the IOException and return null here instead.

Interestingly, the error ends up the same whether we catch and return null or let it bubble up, which is something like this:
> [error] Schema Definition Error: Error loading schema due to /this/path/does/not/exist (No such file or directory)